crates/cext/src/transpiler/passes/commutative_cancellation.rs

```
@@ -0,0 +1,124 @@
+/// pointed to will be updated with the modified circuit if the pass is able to remove any gates.
+/// @param target The target for the CommutativeCancellation pass. The target is used to specify
+/// as fallback Z rotation gate to use for cancellations if one can't be identified in the circuit.
+/// This can be a null pointer which indicates there isn't a target to use.
+///
+/// @returns The integer return code where 0 represents no error and 1 is
+/// used to indicate an error was encountered during the execution of the pass.
+///
+/// # Example
+///
+/// ```c
+///     QkCircuit *qc = qk_circuit_new(4, 0);
+///     uint_t cx_qargs[2] = {0, 1};
+///     qk_circuit_gate(qc, QkGate_CX, cx_qargs, NULL);
+///     qk_circuit_gate(qc, QkGate_Z, &0, NULL);
+///     qk_circuit_gate(qc, QkGate_CX, cx_qargs, NULL);
+///
+///     int return = qk_transpiler_pass_standalone_commutative_cancellation(qc, NULL, 1.0);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` or ``target`` is not a valid, ``QkCircuit`` and ``QkTarget``.
+/// ``QkCircuit`` is not expected to be null and behavior is undefined if it is.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_transpiler_pass_standalone_commutative_cancellation(
+    circuit: *mut CircuitData,
+    target: *const Target,
+    approximation_degree: f64,
+) -> i32 {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let circuit = unsafe { mut_ptr_as_ref(circuit) };
+    let target = if target.is_null() {
+        None
+    } else {
+        Some(unsafe { const_ptr_as_ref(target) })
+    };
+    let mut dag = match DAGCircuit::from_circuit_data(circuit, false, None, None, None, None) {
+        Ok(dag) => dag,
+        Err(e) => panic!("{}", e),
+    };
+    if !(0.0..=1.0).contains(&approximation_degree) {
+        panic!("Invalid value provided for approximation degree, only NAN or values between 0.0 and 1.0 inclusive are valid");
+    }
```

Tiny suggestion, but how about moving the validation to the top of this function so it would fail fast and before doing expensive conversions from circuit to dag?

---

Done in: https://github.com/Qiskit/qiskit/pull/14831/commits/88d6958a36e3e66c424e7d393a47785354387c1d